core/templates/domain/improvements/ExplorationTaskObjectFactory.ts

```diff
@@ -113,8 +120,9 @@ export class ExplorationTaskObjectFactory {
       case 'successive_incorrect_answers':
         return this.siaTaskObjectFactory.createFromBackendDict(backendDict);
       default: {
-        const invalidTaskType: never = backendDict.task_type;
-        throw new Error('unsupported task type: ' + invalidTaskType);
+        const invalidBackendDict: never = backendDict;
+        throw new Error(
+          'unsupported task type: ' + JSON.stringify(invalidBackendDict));
```

Comment:
Perhaps specify the problematic task type as well?

Reply:
Yup, that's already included as part of the backend dict.

Reply:
Done.